File: wire-android-automation/common/src/main/java/com/wearezeta/auto/common/email/messages/ActivationMessage.java
package com.wearezeta.auto.common.email.messages;

import java.util.concurrent.Future;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

import com.wearezeta.auto.common.misc.Timedelta;

public class ActivationMessage extends WireMessage {

	public static final Timedelta ACTIVATION_TIMEOUT = Timedelta.ofSeconds(120); // seconds

	public ActivationMessage(String msg) throws Exception {
		super(msg);
	}

    public String getXZetaKey() {
        return this.getHeaderValue(ZETA_KEY_HEADER_NAME);
    }

    public String getXZetaCode() {
        return this.getHeaderValue(ZETA_CODE_HEADER_NAME);
    }

	public String getActivationLink() throws Exception {
		String regex = "https://[a-zA-Z_0-9.=-]+/verify/\\?key=([a-zA-Z_0-9=-]+)&code=([0-9]+)";
		Pattern p = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
		Matcher urlMatcher = p.matcher(this.getContent());
		if (urlMatcher.find()) {
			return urlMatcher.group(0);
		} else {
			throw new Exception("Unable to extract URL from mail: " + getContent());
		}
	}

	public static String getMessageContent(Future<String> activationMessage) throws Exception {
		ActivationMessage sentence = new ActivationMessage(activationMessage.get());
		return sentence.getContent();
	}

	public static final String MESSAGE_PURPOSE = "Activation";

	@Override
	protected String getExpectedPurposeValue() {
		return MESSAGE_PURPOSE;
	}
}

File: tests/testsCore/src/main/kotlin/com/wire/android/tests/core/messages/ActivationMessage.kt
package com.wire.android.tests.core.messages

import com.wire.android.tests.core.utils.Timedelta
import java.util.concurrent.Future
import java.util.regex.Pattern

class ActivationMessage(msg: String) : WireMessage(msg) {

    companion object {
        val ACTIVATION_TIMEOUT: Timedelta = Timedelta.ofSeconds(120.0) // seconds
        const val MESSAGE_PURPOSE = "Activation"
        private const val ACTIVATION_LINK_REGEX = "https://[a-zA-Z_0-9.=-]+/verify/\\?key=([a-zA-Z_0-9=-]+)&code=([0-9]+)"
    }

    val xZetaKey: String?
        get() = getHeaderValue(ZETA_KEY_HEADER_NAME)

    val xZetaCode: String?
        get() = getHeaderValue(ZETA_CODE_HEADER_NAME)

    @Throws(Exception::class)
    fun getActivationLink(): String {
        val pattern = Pattern.compile(ACTIVATION_LINK_REGEX, Pattern.CASE_INSENSITIVE)
        val matcher = pattern.matcher(content)
        return if (matcher.find()) {
            matcher.group(0)
        } else {
            throw Exception("Unable to extract URL from mail: $content")
        }
    }

   // companion object {
        @Throws(Exception::class)
        fun getMessageContent(activationMessage: Future<String>): String {
            val sentence = ActivationMessage(activationMessage.get())
            return sentence.content
        }
   // }

    override val expectedPurposeValue: String
        get() = MESSAGE_PURPOSE
}
